src/output_writers.jl

```diff
@@ -270,126 +270,215 @@ end
-Base.@kwdef mutable struct NetCDFOutputWriter <: AbstractOutputWriter
-             dir  :: AbstractString = "."
-          prefix  :: AbstractString = ""
-        frequency :: Int    = 1
-          padding :: Int    = 9
-    naming_scheme :: Symbol = :iteration
-      compression :: Int    = 3
-            async :: Bool   = false
-end
+"""
+    netcdf_spatial_dimensions(::field)
 
-ext(fw::NetCDFOutputWriter) = ".nc"
+Returns the dimensions associated with a field.
 
-function filename(fw, name, iteration)
-    if fw.naming_scheme == :iteration
-        fw.prefix * name * lpad(iteration, fw.padding, "0") * ext(fw)
-    elseif fw.naming_scheme == :file_number
-        file_num = Int(iteration / fw.frequency)
-        fw.prefix * name * lpad(file_num, fw.padding, "0") * ext(fw)
-    else
-        throw(ArgumentError("Invalid naming scheme: $(fw.naming_scheme)"))
-    end
+Examples
+========
+
+julia> netcdf_spatial_dimensions(model.velocities.u)
+("xF", "yC", "zC")
+
+julia> netcdf_spatial_dimensions(model.tracers.T)
+("xC", "yC", "zC")
+"""
+function netcdf_spatial_dimensions(field::Field{LX, LY, LZ}) where {LX, LY, LZ}
+    xdim(LX), ydim(LY), zdim(LZ)
 end
 
-function write_output(model::Model, fw::NetCDFOutputWriter)
-    fields = Dict(
+xdim(::Type{Face}) = "xF"
+xdim(::Type{Cell}) = "xC"
+ydim(::Type{Face}) = "yF"
+ydim(::Type{Cell}) = "yC"
+zdim(::Type{Face}) = "zF"
+zdim(::Type{Cell}) = "zC"
+
+# This function converts and integer to a range, and nothing to a Colon
+get_slice(n::Integer) = n:n
+get_slice(n::UnitRange) = n
+get_slice(n::Nothing) = Colon()
+
+"""
+    write_grid(model; filename="grid.nc", mode="c", 
+               compression=0, attrib=Dict(), slice_kw...)
+
+Writes a grid.nc file that contains all the dimensions of the domain.
+
+Keyword arguments
+=================
+
+    - `filename::String`  : File name to be saved under
+    - `mode::String`      : Netcdf file is opened in either clobber ("c") or append ("a") mode (Default: "c" )
+    - `compression::Int`  : Defines the compression level of data (0-9, default 0)
+    - `attrib::Dict`      : Attributes (default: Dict())
+"""
+function write_grid(model; filename="./grid.nc", mode="c",
+                    compression=0, attrib=Dict(), slice_kw...)
+    dimensions = Dict(
         "xC" => collect(model.grid.xC),
         "yC" => collect(model.grid.yC),
         "zC" => collect(model.grid.zC),
-        "xF" => collect(model.grid.xF),
-        "yF" => collect(model.grid.yF),
-        "zF" => collect(model.grid.zF),
-        "u" => Array(parentdata(model.velocities.u)),
-        "v" => Array(parentdata(model.velocities.v)),
-        "w" => Array(parentdata(model.velocities.w)),
-        "T" => Array(parentdata(model.tracers.T)),
-        "S" => Array(parentdata(model.tracers.S))
+        "xF" => collect(model.grid.xF)[1:end-1],
+        "yF" => collect(model.grid.yF)[1:end-1],
+        "zF" => collect(model.grid.zF)[1:end-1]
+    )
+    dim_attrib = Dict(
```

Comment:
Can we maybe add a keyword argument `grid_units` with a default of `meters`? That way the user can specify correct units if they need to. It's an incomplete solution but at least provides the possibility of correct behavior.